org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/views/IssueLocationsView.java

```diff
@@ -0,0 +1,402 @@
+      if (element instanceof RootNode) {
+        return SonarLintImages.IMG_ISSUE;
+      }
+      return super.getImage(element);
+    }
+
+  }
+
+  public void setInput(@Nullable IMarker sonarlintMarker) {
+    if (sonarlintMarker == null) {
+      clearInput();
+      return;
+    }
+    // Find IFile and open Editor
+    // Super defensing programming because we don't really understand what is initialized at startup (SLE-122)
+    IWorkbenchWindow window = PlatformUI.getWorkbench().getActiveWorkbenchWindow();
+    if (window == null) {
+      showMarkerWithNoEditor(sonarlintMarker);
+      return;
+    }
+    IWorkbenchPage page = window.getActivePage();
+    if (page == null) {
+      showMarkerWithNoEditor(sonarlintMarker);
+      return;
+    }
+    for (IEditorReference editor : page.getEditorReferences()) {
+      IEditorInput editorInput;
+      try {
+        editorInput = editor.getEditorInput();
+      } catch (PartInitException e) {
+        SonarLintLogger.get().error("Unable to restore editor", e);
+        continue;
+      }
+      // Cast needed for older Eclipse versions
+      IFile file = (IFile) editorInput.getAdapter(IFile.class);
+      if (file != null && sonarlintMarker.getResource().equals(file)) {
+        IEditorPart editorPart = editor.getEditor(false);
+        if (editorPart instanceof ITextEditor) {
+          this.currentEditor = (ITextEditor) editorPart;
+          this.currentDoc = currentEditor.getDocumentProvider().getDocument(editorPart.getEditorInput());
+          this.currentFile = file;
+          locationsViewer.setInput(sonarlintMarker);
+          try {
+            ShowIssueFlowsMarkerResolver.toggleAnnotations((IMarker) locationsViewer.getInput(), currentEditor, true);
+          } catch (BadPositionCategoryException e) {
+            SonarLintLogger.get().error("Unable to show annotations", e);
+          }
+          return;
+        }
+      }
+    }
+    showMarkerWithNoEditor(sonarlintMarker);
+  }
+
+  private void showMarkerWithNoEditor(IMarker sonarlintMarker) {
+    clearCurrentEditor();
+    locationsViewer.setInput(sonarlintMarker);
+  }
+
+  private void clearInput() {
+    clearCurrentEditor();
+    locationsViewer.setInput(null);
+  }
+
+  private void clearCurrentEditor() {
+    this.currentEditor = null;
+    this.currentDoc = null;
+    this.currentFile = null;
+  }
+
+  @Override
+  public void selectionChanged(IWorkbenchPart part, ISelection selection) {
+    IMarker selectedMarker = AbstractSonarWebView.findSelectedSonarIssue(selection);
+    if (selectedMarker != null && selectedMarker != locationsViewer.getInput()) {
+      setInput(selectedMarker);
+    }
+  }
+
+  private void startListeningForSelectionChanges() {
+    getSite().getPage().addPostSelectionListener(this);
+  }
+
+  private void stopListeningForSelectionChanges() {
+    getSite().getPage().removePostSelectionListener(this);
+  }
+
+  @Override
+  public void createPartControl(Composite parent) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 16 to the 15 allowed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3776)